libs/alerts/src/main/java/org/codice/ddf/system/alerts/SystemNotice.java

```diff
@@ -0,0 +1,175 @@
+
+    //Key for the Set<String> of details of this notice
+    public static final String SYSTEM_NOTICE_DETAILS_KEY = "details";
+
+    public SystemNotice() {
+        put(EVENT_TYPE_KEY, SYSTEM_NOTICE_EVENT_TYPE);
+        setTime(Date.from(Instant.now()));
+        setId(UUID.randomUUID()
+                .toString());
+        setSource("Unknown");
+        setPriority(NoticePriority.NORMAL.value());
+        setTitle("");
+        setDetails(new HashSet<>());
+
+        try {
+            setHostAddress(InetAddress.getLocalHost()
+                    .getHostAddress());
+            setHostName(InetAddress.getLocalHost()
+                    .getHostName());
+        } catch (UnknownHostException e) {
+            // Should never happen
+            throw new IllegalStateException(
+                    "Cannot create system notice because the host name could not be retrieved. Reason: "
+                            + e.getMessage());
+        }
+    }
+
+    public SystemNotice(String source, int priority, String title, Set<String> details) {
+        this();
+        setSource(source);
+        setPriority(priority);
+        setTitle(title);
+        setDetails(details == null ? new ArrayList<>() : details);
+
+    }
+
+    public String getId() {
+        return (String) this.get(SYSTEM_NOTICE_ID_KEY);
+    }
+
+    public String getSource() {
+        return (String) this.get(SYSTEM_NOTICE_SOURCE_KEY);
+    }
+
+    public String getHostName() {
+        return (String) this.get(SYSTEM_NOTICE_HOST_NAME_KEY);
+    }
+
+    public String getHostAddress() {
+        return (String) this.get(SYSTEM_NOTICE_HOST_ADDRESS_KEY);
+    }
+
+    public int getPriority() {
+        Object value = this.get(SYSTEM_NOTICE_PRIORITY_KEY);
+        if(value instanceof String){
+            return Integer.parseInt((String)value);
+        }
+        return (int)value;
+    }
+
+    public String getTitle() {
+        return (String) this.get(SYSTEM_NOTICE_TITLE_KEY);
+    }
+
+    public Set<String> getDetails() {
+        Object value = this.get(SYSTEM_NOTICE_DETAILS_KEY);
+        if(value instanceof String){
+            return Collections.singleton((String)value);
+        }
+        return (Set<String>) value;
+    }
+
+    public Date getTime() {
+        Object value = this.get(SYSTEM_NOTICE_TIME_KEY);
+        if(value instanceof Long){
+            return new Date((long)value);
+        }
+        return (Date)value;
+    }
+
+    public void setId(String id) {
+        this.put(SYSTEM_NOTICE_ID_KEY, id);
+    }
+
+    public void setSource(String source) {
+        this.put(SYSTEM_NOTICE_SOURCE_KEY, source);
+    }
+
+    public void setHostName(String host) {
+        this.put(SYSTEM_NOTICE_HOST_NAME_KEY, host);
+    }
+
+    public void setHostAddress(String address) {
+        this.put(SYSTEM_NOTICE_HOST_ADDRESS_KEY, address);
+    }
+
+    public void setPriority(int priority) {
+        this.put(SYSTEM_NOTICE_PRIORITY_KEY, priority);
```

Comment:
❗️If we don't store `NoticePriority` directly in the map (my preference), we should at least validate that the number provided is a valid value.